### DISCUSSION TITLE
Extract.prototype.canvas : using the function   frequently   causes framerate decreasing

### DISCUSSION THREAD

**Discussion by avenir-zhang** [pixijs/pixijs#8062] [Q&A]
### Expected Behavior
In project , I need to axtract image , use the **Extract.prototype.canvas** , when I use the function frequently, the framerate decrease to 1-2, watch the source , the function  uses webgl . readpixels.  why not use cavans.getContex('xxxx').drawImage(canvas)
### Environment

- **`pixi.js` version**: _e.g. 4.7.1_
- **Browser & Version**: _e.g. Chrome 67_
- **OS & Version**: _e.g. Ubuntu 18.04_
- **Running Example**: _url_

**Comment by bigtimebuddy** [comment]
This is a support-related question and I've moved it to discussions.

**Comment by eXponenta** [comment]
```
cavans.getContex('xxxx').drawImage(canvas)
```

require to use `preserveDrawingBuffer` and only canvas related source.

99% cases of extract use a render targets, and it can be greater/less that context.
And this is unsafe, because required to clear canvas and bliting framebuffer to it, that will drop canvas render state.

Anyway a use canvas as texture source require synchronization, and for frequently queries for drawing will be lag.
For Safari webgl canvas as source will produce frame lag - latest flushed to display frame will be drawn, this means that you cannot grab frame between render process to showing on other canvas for fast or sync-important cases.

Conclusion: readPixels is more stable.

But for WebGL 2 a pixel buffers can be used for async low-latencity requests.

readPixels not a problem itself, problems with waiting a GPU tasks finishing.
